### ISSUE TITLE
ReST

### ISSUE THREAD

**Issue by JamesStrabala** [JamesStrabala/Capstone-Resorces#21]
### Title
Reinforced Self-Training (ReST) for Language Modeling
### URL
https://doi.org/10.48550/arXiv.2308.08998
### Summary 
Reinforced Self-Training (ReST) is a method of fine-tuning LM output to ensure their output falls within desired guidelines. In the context of PLMs, this could mean ensuring their products fit a chosen purpose. Human-supervised methods to do this can be fooled by reward hacking.
The ReST training method has two steps with nested loops. The inner loop is the Improve step, where the data set is ranked and filtered based on a reward model of human preferences. The LM is fine-tuned with offline reinforcement learning on the data and the resulting LM policy is used in the next Grow step. In the Grow step, the language model policy generates predictions for different contexts to add to the dataset.
The offline reinforcement learning reduces computational load. The policy's quality is not limited by the original dataset's and it can be easily monitored. The approach is simple with few hyperparameters. 
Experiments found that each successive Grow and Improve step improved performance. Online Reinforcement Learning was found to be as effective as a round of ReST with one Grow step.
### Key Points 
* Rest is a method of reinforcement learning that interatively expands the LM policy.
### Citation
@misc{gulcehre2023reinforcedselftrainingrestlanguage,
      title={Reinforced Self-Training (ReST) for Language Modeling}, 
      author={Caglar Gulcehre and Tom Le Paine and Srivatsan Srinivasan and Ksenia Konyushkova and Lotte Weerts and Abhishek Sharma and Aditya Siddhant and Alex Ahern and Miaosen Wang and Chenjie Gu and Wolfgang Macherey and Arnaud Doucet and Orhan Firat and Nando de Freitas},
      year={2023},
      eprint={2308.08998},
      archivePrefix={arXiv},
      primaryClass={cs.CL},
      url={https://arxiv.org/abs/2308.08998}, 
}
